Branch-SDK/Branch-SDK/Branch.m

```diff
@@ -1048,41 +1050,53 @@ - (NSString *)generateShortUrl:(NSArray *)tags andAlias:(NSString *)alias andTyp
+    NSString *appIdentifier = [BNCPreferenceHelper getBranchKey];
+    if ([appIdentifier isEqualToString:NO_STRING_VALUE]) {
+        appIdentifier = [BNCPreferenceHelper getAppKey];
+    }
     
-    [longURL appendString:[NSString stringWithFormat:@"/a/%@?", [BNCPreferenceHelper getAppKey]]];
+    if ([appIdentifier isEqualToString:NO_STRING_VALUE]) {
+        NSLog(@"No Branch Key specified, cannot create a long url");
+        return nil;
+    }
     
-    NSData *jsonData = [BNCEncodingUtils encodeDictionaryToJsonData:params];
-    NSString *base64EncodedParams = [BNCEncodingUtils base64EncodeData:jsonData];
+    NSString *baseLongUrl = [NSString stringWithFormat:@"%@/a/%@", BNC_LINK_URL, appIdentifier];
+
+    return [self longUrlWithBaseUrl:baseLongUrl params:params tags:tags feature:feature channel:nil stage:stage alias:alias duration:0 type:BranchLinkTypeUnlimitedUse];
+}
+
+- (NSString *)longUrlWithBaseUrl:(NSString *)baseUrl params:(NSDictionary *)params tags:(NSArray *)tags feature:(NSString *)feature channel:(NSString *)channel stage:(NSString *)stage alias:(NSString *)alias duration:(NSUInteger)duration type:(BranchLinkType)type {
+    NSMutableString *longUrl = [[NSMutableString alloc] initWithFormat:@"%@?", baseUrl];
     
-    if (channel == nil) channel = @"";
-    if (feature == nil) feature = @"";
-    if (stage == nil) stage = @"";
-    if (alias == nil) alias = @"";
+    for (NSString *tag in tags) {
+        [longUrl appendFormat:@"tags=%@&", tag];
+    }
     
-    NSMutableArray *urlParamStrings = [@[ @{ @"channel": channel },
-                                @{ @"feature": feature },
-                                @{ @"stage": stage },
-                                @{ @"alias": alias },
-                                 @{ @"data": base64EncodedParams} ] mutableCopy];
+    if ([alias length]) {
+        [longUrl appendFormat:@"alias=%@&", alias];
+    }
     
-    for (NSString *tag in tags) {
-        [urlParamStrings addObject:@{ @"tags": tag}];
+    if ([channel length]) {
+        [longUrl appendFormat:@"channel=%@&", channel];
     }
     
-    for (NSDictionary *thisParam in urlParamStrings) {
-        NSString *paramName = [[thisParam allKeys] firstObject];
-        NSString *paramValue = [[thisParam allValues] firstObject];
-        if (paramValue.length > 0) {
-            if(![longURL hasSuffix:@"?"]) {
-                [longURL appendString:@"&"];
-            }
-            [longURL appendFormat:@"%@=%@", paramName, paramValue];
-        }
+    if ([feature length]) {
+        [longUrl appendFormat:@"feature=%@&", feature];
+    }
+    
+    if ([stage length]) {
+        [longUrl appendFormat:@"stage=%@&", stage];
     }
     
-    return longURL;
+    [longUrl appendFormat:@"type=%lld&", (long long)type];
```

Comment:
Hmm, it looks like that's right. I'm not sure why I thought it was this way. If I can't find anything supporting that, I'll switch to just `%ld`.